frontend/src/types/react-table-config.d.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  UseSortByColumnOptions,
+  UseSortByColumnProps,
+  UseSortByHooks,
+  UseSortByInstanceProps,
+  UseSortByOptions,
+  UseSortByState,
+} from 'react-table'
+
+declare module 'react-table' {
+  // https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react-table
+
+  export interface TableOptions<
+    D extends Record<string, unknown>
+  > extends UseSortByOptions<D>,
+      // note that having Record here allows you to add anything to the options, this matches the spirit of the
+      // underlying js library, but might be cleaner if it's replaced by a more specific type that matches your
+      // feature set, this is a safe default.
+      Record<string, any> {}
+
+  export interface Hooks<
+    D extends Record<string, unknown> = Record<string, unknown>
+  > extends UseSortByHooks<D> {}
+
+  export interface TableInstance<
+    D extends Record<string, unknown> = Record<string, unknown>
+  > extends UseSortByInstanceProps<D> {}
+
+  export interface TableState<
+    D extends Record<string, unknown> = Record<string, unknown>
+  > extends UseSortByState<D> {}
+
+  export interface ColumnInterface<
+    D extends Record<string, unknown> = Record<string, unknown>
+  > extends UseSortByColumnOptions<D> {}
+
+  export interface ColumnInstance<
+    D extends Record<string, unknown> = Record<string, unknown>
+  > extends UseSortByColumnProps<D> {}
+
+  export interface Cell<
+    D extends Record<string, unknown> = Record<string, unknown>,
+    V = any
+  > {}
+
+  export interface Row<
+    D extends Record<string, unknown> = Record<string, unknown>
+  > {}
+}
```

Comment:
Question: do you have a sense of how much updating this file will need?

Reply:
We would need to update these typedefs if we used more of the available plugins. Right now, we only use `useSortBy` (see code below), but I could see us add [`useFilters`](https://react-table.tanstack.com/docs/api/useFilters), for example.

```ts
const {
  getTableProps,
  getTableBodyProps,
  headerGroups,
  rows,
  prepareRow,
} = useTable({ columns, data }, useSortBy) // <- last argument here
```

Reply:
I'm hoping they release v8 of the package soon because it's written entirely in TypeScript. Even then though, I feel like the benefits outweigh the drawbacks.

Reply:
Thanks for explaining, helpful to know!